### ISSUE TITLE
Unsupported SDL2 video driver "x11". Only X11 is supported on Linux.

### ISSUE THREAD

**Issue by FurrFett** [Mr-Wiseguy/Zelda64Recomp#285]
## If you have a crash on startup, please make sure your graphics drivers are up to date before submitting a bug report.

**Describe the bug**
New linux fixed old problem, but now new problem.  

 >Unsupported SDL2 video driver "x11". Only X11 is supported on Linux.

**To Reproduce**
1. Go to '.Zelda64Recompiled.'
2. run or run in Konsole/terminal.

**Expected behavior**
Application launches and runs

**Screenshots**
![Screenshot_20240526_172500](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/107557387/1d0858e6-51b7-4cd1-9178-8ef913c710f7)

**Desktop (please complete the following information):**
 Operating System: Linux Mint 21.3
KDE Plasma Version: 5.24.7
KDE Frameworks Version: 5.92.0
Qt Version: 5.15.3
Kernel Version: 6.8.5-060805-generic (64-bit)
Graphics Platform: X11
Processors: 16 × AMD Ryzen 7 5800X3D 8-Core Processor
Memory: 31.3 GiB of RAM
Graphics Processor: AMD Radeon RX 6600 XT

**Comment by Reonu** [comment]
Hello, when the artifact for this PR finishes building, could you try it out? It will probably give you an error as well, but I'm interested in what the error says. It'll give us useful information. Thanks.

https://github.com/Zelda64Recomp/Zelda64Recomp/pull/286

**Comment by FurrFett** [comment]
> Hello, when the artifact for this PR finishes building, could you try it out? It will probably give you an error as well, but I'm interested in what the error says. It'll give us useful information. Thanks.
> 
> #286

yeah sure, no problem! I'm still learning to use linux as my only daily driver. so as long as its not too difficult i dont have a problem helping.

**Comment by Reonu** [comment]
> > Hello, when the artifact for this PR finishes building, could you try it out? It will probably give you an error as well, but I'm interested in what the error says. It'll give us useful information. Thanks.
> > #286
> 
> yeah sure, no problem! I'm still learning to use linux as my only daily driver. so as long as its not too difficult i dont have a problem helping.

Go to the PR I linked. Go to "Checks". From there go to "Artifacts". Download the "Linux-Release" artifact. That's a new build. Run the build like you were running the other one, and post the error you get with it here.

**Comment by FurrFett** [comment]
![image](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/107557387/b41d7aed-b8e3-4d66-bbde-07af27845801)

**Comment by Reonu** [comment]
Thank you. I just pushed another another change to that PR with a **potential** dumb fix that I came up with. I'm not sure if it'll work. In any case, when the artifact is ready (it takes around 5 minutes to build), please download the new artifact (it's in the same location) and test again. Let me know if it works.

If it doesn't work, here's a possible workaround: 
Add Zelda64Recompiled to Steam as a non-Steam game. Right click it and go to "properties". In "compatibility" enable forcing a specific compatibility tool and set it to "Steam Linux Runtime". Then launch it through steam. This should work. But before you try this - please try the new artifact, I really want to know if it solves the issue.

Thanks.

**Comment by FurrFett** [comment]
nope, just checked it.  and it's back to the original error i was getting in my first post.       I've used it by forcing the proton compatibility in this steam prior to this.  but, i saw there was a new release with fixes so i figured i'd give it a shot and then see if i could help by posting the new error. 

guess it still need work.   maybe my simple brain, but is... the simple fix somewhere in the ini (or equivalent file) trying to call 'x11' when it should be calling for 'X11' ?   i doubt its that simple, but i wonder if a simple capital letter somewhere is causing the problem?

**Comment by Reonu** [comment]
We'll investigate this further. Thank you for the help.

**Comment by Reonu** [comment]
@dcvz any chance you could look into this? `wmInfo-subsystem` seems to be returning `SDL_SYSWM_UNKNOWN` despite the user being on X11, so (unless I'm misunderstanding) Recomp things the environment is not X11, when it actually is, and closes itself.

**Comment by dcvz** [comment]
> @dcvz any chance you could look into this? `wmInfo-subsystem` seems to be returning `SDL_SYSWM_UNKNOWN` despite the user being on X11, so (unless I'm misunderstanding) Recomp things the environment is not X11, when it actually is, and closes itself. 

Yup can take a look later today!

**Comment by Reonu** [comment]
Thank you!

**Comment by dcvz** [comment]
@FurrFett just for testing sake, do you get the same issue when using the AppImage? You can find them here: https://github.com/Zelda64Recomp/Zelda64Recomp/releases/tag/v1.1.0

**Comment by jcmfernandes** [comment]
> @FurrFett just for testing sake, do you get the same issue when using the AppImage? You can find them here: https://github.com/Zelda64Recomp/Zelda64Recomp/releases/tag/v1.1.0

I can confirm that I get the same error using the Linux build on PopOS 22.04 (an Ubuntu derivative), but it just fine using the AppImage.

**Comment by FurrFett** [comment]
Im working at the moment (truck driver) i dunno when ill get a chance to check again. Either tomorrow morning or tomorrow night but i will check. 

Ive never used an app image (guessing its like a flatpak or something?). But ill give it a try when i can.

**Comment by shizukachan** [comment]
Ubuntu 22.04's libSDL is probably too old or not compiled with the right config options. I eventually found a workable solution was to follow the CI config:

https://github.com/Zelda64Recomp/Zelda64Recomp/blob/dev/.github/workflows/validate.yml#L51

You will need to build SDL, and point your recomp binary to use the newly built SDL (or do a `sudo make install` for SDL.) You will need to make sure your `./configure` output shows `X11(dynamic)` as an output option; building `libsdl2-dev` as done in the CI will get you a SDL build that supports everything that it should.

**Comment by WinterWolfGames** [comment]
I'm having this issue but only when launching from lutris for some reason. I am able to launch the appimage just fine on its own and with gamemode through terminal. I have no idea why lutris specifically cannot launch this. all of my drivers are (to my knowledge) up-to-date.

here's my log from lutris

Started initial process 43845 from gamemoderun ./Zelda64Recompiled-x86_64.AppImage
Start monitoring process.
gamemodeauto: 
gamemodeauto: 
gamemodeauto: 
Unsupported SDL2 video driver "x11". Only X11 is supported on Linux.
Monitored process exited.
Initial process has exited (return code: 256)
All processes have quit
Exit with return code 256

**Comment by gordoncicero** [comment]
> Ubuntu 22.04's libSDL is probably too old or not compiled with the right config options. I eventually found a workable solution was to follow the CI config:
> 
> https://github.com/Zelda64Recomp/Zelda64Recomp/blob/dev/.github/workflows/validate.yml#L51
> 
> You will need to build SDL, and point your recomp binary to use the newly built SDL (or do a `sudo make install` for SDL.) You will need to make sure your `./configure` output shows `X11(dynamic)` as an output option; building `libsdl2-dev` as done in the CI will get you a SDL build that supports everything that it should.

when i tried using the sudo make install sdl it says to stop, "No rule to make target install. stop" im really new to linux and idk how stuff work so sorry in any case

**Comment by gordoncicero** [comment]
> Thank you. I just pushed another another change to that PR with a **potential** dumb fix that I came up with. I'm not sure if it'll work. In any case, when the artifact is ready (it takes around 5 minutes to build), please download the new artifact (it's in the same location) and test again. Let me know if it works.
> 
> If it doesn't work, here's a possible workaround: Add Zelda64Recompiled to Steam as a non-Steam game. Right click it and go to "properties". In "compatibility" enable forcing a specific compatibility tool and set it to "Steam Linux Runtime". Then launch it through steam. This should work. But before you try this - please try the new artifact, I really want to know if it solves the issue.
> 
> Thanks.

thanks mate, the linux solution really helped

**Comment by gordoncicero** [comment]
> Thank you. I just pushed another another change to that PR with a **potential** dumb fix that I came up with. I'm not sure if it'll work. In any case, when the artifact is ready (it takes around 5 minutes to build), please download the new artifact (it's in the same location) and test again. Let me know if it works.
> 
> If it doesn't work, here's a possible workaround: Add Zelda64Recompiled to Steam as a non-Steam game. Right click it and go to "properties". In "compatibility" enable forcing a specific compatibility tool and set it to "Steam Linux Runtime". Then launch it through steam. This should work. But before you try this - please try the new artifact, I really want to know if it solves the issue.
> 
> Thanks.

hey, tried, and the start menu works perfecly, but for some reason when i click play game the game closes
